sdk/react/panels/cast/CastConnectedScreen.js

```diff
@@ -0,0 +1,345 @@
+    height: PropTypes.number.isRequired,
+    volume: PropTypes.number.isRequired,
+    fullscreen: PropTypes.bool.isRequired,
+    cuePoints: PropTypes.arrayOf(PropTypes.double).isRequired,
+    stereoSupported: PropTypes.bool.isRequired,
+    multiAudioEnabled: PropTypes.bool.isRequired,
+    playbackSpeedEnabled: PropTypes.bool.isRequired,
+    selectedPlaybackSpeedRate: PropTypes.string.isRequired,
+    handlers: PropTypes.shape({
+      onSwitch: PropTypes.func,
+      onPress: PropTypes.func,
+      onScrub: PropTypes.func,
+      handleVideoTouchStart: PropTypes.func,
+      handleVideoTouchMove: PropTypes.func,
+      handleVideoTouchEnd: PropTypes.func,
+      handleControlsTouch: PropTypes.func,
+      handleShowControls: PropTypes.func,
+    }).isRequired,
+    screenReaderEnabled: PropTypes.bool.isRequired,
+    availableClosedCaptionsLanguages: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+    config: PropTypes.object.isRequired,
+    localizableStrings: PropTypes.object.isRequired,
+    locale: PropTypes.string.isRequired,
+    playing: PropTypes.bool.isRequired,
+    loading: PropTypes.bool.isRequired,
+    initialPlay: PropTypes.bool.isRequired,
+    onDisconnect: PropTypes.func.isRequired,
+    deviceName: PropTypes.string.isRequired,
+    inCastMode: PropTypes.bool.isRequired,
+    previewUrl: PropTypes.string.isRequired,
+  };
+
+  onSeekPressed(skipCountValue) {
+    if (skipCountValue === 0) {
+      return;
+    }
+    const { props } = this;
+    const { playhead, duration } = props;
+    const { skipForwardTime, skipBackwardTime } = props.config.castControls;
+
+    let configSeekValue = (skipCountValue > 0) ? skipForwardTime : skipBackwardTime;
+
+    configSeekValue = Utils.restrictSeekValueIfNeeded(configSeekValue);
+    const seekValue = configSeekValue * skipCountValue;
+    let resultedPlayhead = playhead + seekValue;
+    if (resultedPlayhead < 0) {
+      resultedPlayhead = 0;
+    } else if (resultedPlayhead > duration) {
+      resultedPlayhead = duration;
+    }
+    const resultedPlayheadPercent = duration === 0 ? 0 : resultedPlayhead / duration;
+
+    this.handleScrub(resultedPlayheadPercent);
+  }
+
+  onSwitchPressed(isForwardSwitch) {
+    const { props } = this;
+    const { onSwitch } = props.handlers;
+    onSwitch(isForwardSwitch);
+  }
+
+  generateLiveObject() {
+    const {
+      live, playhead, duration, locale, localizableStrings,
+    } = this.props;
+    if (live) {
+      const isLive = playhead >= duration * VALUES.LIVE_THRESHOLD;
+      return ({
+        label: isLive ? Utils.localizedString(locale, 'LIVE', localizableStrings)
+          : Utils.localizedString(locale, 'GO LIVE', localizableStrings),
+        onGoLive: isLive ? null : this.onGoLive,
+      });
+    }
+    return null;
+  }
+
+  placeholderTapHandler(event) {
+    const { props } = this;
+    const { screenReaderEnabled } = props;
+    const { handleVideoTouchEnd } = props.handlers;
```

Comment:
Why it's different in different functions?

Reply:
Because here we use props directly.

Reply:
const { handlers, screenReaderEnabled } = this.props;
const { handleVideoTouchEnd } = handlers;

Reply:
fixed